Sources/MapboxNavigation/NavigationMapView.swift

```diff
@@ -1239,64 +1239,30 @@ open class NavigationMapView: UIView {
         shapeLayer.textFont = .constant(self.routeDurationAnnotationFontNames)
 
         try style.addLayer(shapeLayer)
+        
+        shapeLayer.symbolSortKey = .expression(Exp(.get) {
+            "sortOrder"
+        })
 
-        let symbolSortKeyString =
-        """
-        ["get", "sortOrder"]
-        """
-
-        if let expressionData = symbolSortKeyString.data(using: .utf8),
-           let expJSONObject = try? JSONSerialization.jsonObject(with: expressionData, options: []) {
-            let property = "symbol-sort-key"
-            try style.setLayerProperty(for: routeDurationAnnotationsLayerIdentifier,
-                                       property: property,
-                                       value: expJSONObject)
-        }
-
-        let expressionString =
-        """
-        [
-          "match",
-          ["get", "tailPosition"],
-          [0],
-          "bottom-left",
-          [1],
-          "bottom-right",
-          "center"
-        ]
-        """
-
-        if let expressionData = expressionString.data(using: .utf8),
-           let expJSONObject = try? JSONSerialization.jsonObject(with: expressionData, options: []) {
-            try style.setLayerProperty(for: routeDurationAnnotationsLayerIdentifier,
-                                       property: "icon-anchor",
-                                       value: expJSONObject)
-            try style.setLayerProperty(for: routeDurationAnnotationsLayerIdentifier,
-                                       property: "text-anchor",
-                                       value: expJSONObject)
+        let anchorExpression = Exp(.match) {
+            Exp(.get) { "tailPosition" }
+            0
+            "bottom-left"
+            1
+            "bottom-right"
+            "center"
         }
-
-        let offsetExpressionString =
-            """
-        [
-          "match",
-          ["get", "tailPosition"],
-          [0],
-          ["literal", [0.5, -1]],
-          ["literal", [-0.5, -1]]
-        ]
-        """
-        
-        if let expressionData = offsetExpressionString.data(using: .utf8),
-           let expJSONObject = try? JSONSerialization.jsonObject(with: expressionData, options: []) {
-            try style.setLayerProperty(for: routeDurationAnnotationsLayerIdentifier,
-                                       property: "icon-offset",
-                                       value: expJSONObject)
-            
-            try style.setLayerProperty(for: routeDurationAnnotationsLayerIdentifier,
-                                       property: "text-offset",
-                                       value: expJSONObject)
+        shapeLayer.iconAnchor = .expression(anchorExpression)
+        shapeLayer.textAnchor = .expression(anchorExpression)
+        
+        let offsetExpression = Exp(.match) {
```

Comment:
At a first glance looks good. But since I am (and probably others) are not too familiar with the new expressions syntax I wonder if it'd be possible to create unit-tests which could help to understand the new logic better and verify whether expressions syntax works as expected. For example:
- Create `NavigationMapView` instance.
- Add route duration annotation layer with certain features (e.g. which contains `tailPosition` feature).
- Verify that layer property, which is based on this feature is correct (e.g. `iconAnchor`).
- Change feature value (e.g. change `tailPosition` from `.left` to `.right`).
- Verify that layer property changes appropriately.

We can consider this as a tail work though.

Reply:
Thanks for the suggestions, I've added the route duration tests for the `tailPosition ` feature and layer property.